src/plugins/intel_cpu/src/nodes/reorder.cpp

```diff
@@ -457,9 +457,24 @@ void Reorder::reorderData(const IMemory &input, const IMemory &output, MultiCach
 
         auto srcMemory = input.getPrimitive();
         auto dstMemory = output.getPrimitive();
+
+        auto srcMemoryDesc = srcMemory.get_desc();
+        auto dstMemoryDesc = dstMemory.get_desc();
+
         auto engine = dstMemory.get_engine();
+
+        if (srcMemoryDesc.get_ndims() != dstMemoryDesc.get_ndims()) {
+            //rank mismatch, try to reshape source mem descriptor
+            constexpr bool allowEmpty = true;
+            auto reshapedSrcMemDesc = srcMemoryDesc.reshape(dstMemoryDesc.get_dims(), allowEmpty);
+            if (reshapedSrcMemDesc) {
+                srcMemoryDesc = reshapedSrcMemDesc;
+                srcMemory = dnnl::memory(srcMemoryDesc, engine, srcMemory.get_data_handle());
+            }
+        }
+
         // try directly reorder
-        reorder = getReorderPrim(cache, dstMemory.get_engine(), srcMemory.get_desc(), dstMemory.get_desc());
+        reorder = getReorderPrim(cache, engine, srcMemoryDesc, dstMemoryDesc);
```

Comment:
When will this happen? Is it safe to simply reshape source mem desc with destination's?

Reply:
It may happen for example when ReadValue output shape is `{2, 1, 2}`, but Assign node input shape `{2, 2}`. Ngraph allows  such combinations. Yep, it's safe as long as such a reshape is allowed, since it doen't require tossing elements in memory.

Reply:
When is a Reorder need for this ReadValue/Assign pair?

Reply:
The memory state tensor may have only one shape either {2, 1, 2} or {2, 2} , it may not have both at the same time, thus when either ReadValue or Assign edge is processed, a `load()` call happens, thus the reorder is invoked. Also this is useful in a general case, when we have memory objects of shape {2, 1, 2} and {2, 2}. Having this change we can conveniently call load, without additional checks and memory desc conversions.